components/resource-creation/ResourceDisplay.tsx

```diff
@@ -84,6 +86,70 @@ function ResourceDisplay() {
     setSelectedDataRequirement({ name: '', content: null });
   };
 
+  const dateForResource = (resource: any) => {
+    if (!resource || !PrimaryDatePaths?.parsedPrimaryDatePaths) {
+      throw new Error('Error: Invalid resource or PrimaryDatePaths');
+    }
+
+    const dateinfo =  PrimaryDatePaths.parsedPrimaryDatePaths[resource.resourceType] 
+    if (!dateinfo) {
+      throw new Error('Error: No date information found for the resourceType');
+    }
+
+    for (const nameOfResourceDate of Object.keys(dateinfo)) {
+      // If only one dataType
+      if(dateinfo[nameOfResourceDate].dataTypes.length == 1 && resource[nameOfResourceDate]){
+        // If the only dataType is a period 
+        if(dateinfo[nameOfResourceDate].dataTypes[0] == 'Period'){
+          return formatPeriod(resource[nameOfResourceDate], nameOfResourceDate)
+        }
+        // If the only dataType is either dateTime or date
+        else {
+          return (
+            <Tooltip arrowPosition='side' arrowOffset={25} arrowSize={8} label={'Date Type: ' + nameOfResourceDate} withArrow position='top-start'>
+              <Text >{formatDate(resource[nameOfResourceDate])}</Text>
+            </Tooltip>  
+          )
+        }
+      }
+      
+      // If isChoiceType is true
+      else {
+        for (const dataType of dateinfo[nameOfResourceDate].dataTypes) {
+          // Capitalize the first char of dataType and append for proper resource date name
+          const fullResourceDateName = nameOfResourceDate + dataType.charAt(0).toUpperCase() + dataType.slice(1);
+          if (resource[fullResourceDateName]) {
+            if (dataType == 'Period') {
+              return formatPeriod(resource[fullResourceDateName], fullResourceDateName);
+            }
+            // Else if dataType == 'dateTime' || 'Date'
+            return (
+              <Tooltip arrowPosition='side' arrowOffset={25} arrowSize={8} label={'Date Type: ' + fullResourceDateName} withArrow position='top-start'>
+                <Text>{formatDate(resource[fullResourceDateName])}</Text>
+              </Tooltip>
+            )
+          }
+        }
+      }
+    }
+    return 'No Date Found'
+  }
+
+  // Formatter for if the date is a period
+  const formatPeriod = (resourcePeriod: any, dateTypeName: string) => {
```

Comment:
```suggestion
  const formatPeriod = (resourcePeriod: fhir4.Period, dateTypeName: string) => {
```
You can make use of the `fhir4` TS types here. Using these will also show you that you'll need to handle the fact that `start` and `end` may not be defined to note a period that doesn't end (ex AllergyIntollerance) or have a known start time.